### PR TITLE
[client] Fix exit node menu not refreshing on Windows

### PR DESCRIPTION
**Pull request by mlsmaycon**
## Describe your changes
TrayOpenedCh is not implemented in the systray library on Windows, so exit nodes were never refreshed after the initial connect. Combined with the management sync not having populated routes yet when the Connected status fires, this caused the exit node menu to remain empty permanently after disconnect/reconnect cycles.

Add a background poller on Windows that refreshes exit nodes while connected, with fast initial polling to catch routes from management sync followed by a steady 10s interval. On macOS/Linux, TrayOpenedCh continues to handle refreshes on each tray open.

Also fix a data race on connectClient assignment in the server's connect() method and add nil checks in CleanState/DeleteState to prevent panics when connectClient is nil.
### Checklist
- [x] Is it a bug fix
- [ ] Is a typo/documentation fix
- [ ] Is a feature enhancement
- [ ] It is a refactor
- [ ] Created tests that fail without the change (if possible)

> By submitting this pull request, you confirm that you have read and agree to the terms of the [Contributor License Agreement](https://github.com/netbirdio/netbird/blob/main/CONTRIBUTOR_LICENSE_AGREEMENT.md).

## Documentation
Select exactly one:

- [ ] I added/updated documentation for this change
- [x] Documentation is **not needed** for this change (explain why)

### Docs PR URL (required if "docs added" is checked)
Paste the PR link from https://github.com/netbirdio/docs here:

https://github.com/netbirdio/docs/pull/__

## Summary by CodeRabbit

* **Bug Fixes**
  * Prevented nil-pointer issues by guarding connection state checks
  * Ensured cleanup clears connection state safely and cancels exit-node retries on disconnect
  * Improved concurrent access safety for connection client

* **New Features**
  * OS-aware exit-node refresh with optimized polling and more reliable menu refresh
  * Safer connection initialization and assignment flow

* **Tests**
  * Added tests covering connection synchronization, concurrent access, nil/edge cases, and exit-node refresh behavior

### CODE REVIEW

File: client/ui/network.go [hunk 1]
```diff
@@ -335,16 +329,75 @@ func (s *serviceClient) updateNetworksBasedOnDisplayTab(tabs *container.AppTabs,
...
+func (s *serviceClient) startExitNodeRefresh() {
+	s.cancelExitNodeRetry()
+
+	if runtime.GOOS == "windows" {
+		ctx, cancel := context.WithCancel(s.ctx)
+		s.exitNodeMu.Lock()
+		s.exitNodeRetryCancel = cancel
+		s.exitNodeMu.Unlock()
+
+		go s.pollExitNodes(ctx)
```
[the remark this answers is not in the document]
Comment (mlsmaycon) [reply client/ui/network.go:345]: cancelExitNodeRetry holds the log too

**Review by lixmal — APPROVED** [review]

File: client/ui/network.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (mlsmaycon) [reply client/ui/network.go:345]: it doesn't release before: 
```Go
func (s *serviceClient) cancelExitNodeRetry() {
	s.exitNodeMu.Lock()
	if s.exitNodeRetryCancel != nil {
		s.exitNodeRetryCancel()
		s.exitNodeRetryCancel = nil
	}
	s.exitNodeMu.Unlock()
}
```
